src/solo12.cpp

```diff
@@ -0,0 +1,219 @@
+#include <cmath>
+#include "real_time_tools/spinner.hpp"
+#include "blmc_robots/common_programs_header.hpp"
+#include "blmc_robots/solo12.hpp"
+
+namespace blmc_robots{
+
+const double Solo12::max_joint_torque_security_margin_ = 0.99;
+
+Solo12::Solo12()
+{
+  /**
+   * Hardware properties
+   */
+  motor_inertias_.setZero();
+  motor_torque_constants_.setZero();
+  joint_gear_ratios_.setZero();
+  motor_max_current_.setZero();
+  max_joint_torques_.setZero();
+  joint_zero_positions_.setZero();
+  reverse_polarities_.fill(false);
+
+  /**
+   * Hardware status
+   */
+  for(unsigned i=0 ; i<motor_enabled_.size(); ++i)
+  {
+    motor_enabled_[i] = false;
+    motor_ready_[i] = false;
+  }
+  for(unsigned i=0 ; i<motor_board_enabled_.size(); ++i)
+  {
+    motor_board_enabled_[0] = false;
+    motor_board_errors_[0] = 0;
+  }
+
+  /**
+   * Joint data
+   */
+  joint_positions_.setZero();
+  joint_velocities_.setZero();
+  joint_torques_.setZero();
+  joint_target_torques_.setZero();
+  joint_encoder_index_.setZero();
+
+  /**
+   * Additional data
+   */
+  slider_positions_.setZero();
+  contact_sensors_states_.setZero();
+
+  /**
+   * Setup some known data
+   */
+
+  // for now this value is very small but it is currently for debug mode
+  motor_max_current_.fill(4.0); // TODO: set as paramters?
+  motor_torque_constants_.fill(0.025);
+  motor_inertias_.fill(0.045);
+  joint_gear_ratios_.fill(9.0);
+}
+
+void Solo12::initialize(const std::string &network_id)
+{
+  network_id_ = network_id;
+  
+  // Create the different mapping
+  map_joint_id_to_motor_board_id_ = {0, 1, 1, 0, 2, 2, 3, 4, 4, 3, 5, 5};
+  map_joint_id_to_motor_port_id_ = {0, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0};
+
+  // Initialize the communication with the main board.
+  main_board_ptr_ = std::make_shared<MasterBoardInterface>(network_id_);
+  main_board_ptr_->Init();
+
+  // create the SpiBus (blmc_drivers wrapper around the MasterBoardInterface)
+  spi_bus_ = std::make_shared<blmc_drivers::SpiBus>(main_board_ptr_,
+                                                    motor_boards_.size());
+
+  // create the motor board objects:
+  for(size_t mb_id = 0 ; mb_id < motor_boards_.size() ; ++mb_id)
```

Comment:
Please fix formating of the `for` statement. Also below.

Maybe run clang-format in general on this PR?

Reply:
I will progress on the formatting on a different PR. I would need to check the robot behaviors after the general formatting.